ragna/core/_components.py

```diff
@@ -76,6 +92,22 @@ def _protocol_model(cls) -> Type[pydantic.BaseModel]:
         return merge_models(cls.display_name(), *cls._protocol_models().values())
 
 
+@dataclass
+class Embedding:
+    embedding: list[float]
+    chunk: Chunk
+
+
+class EmbeddingModel(Component, ABC):
```

Comment:
Open for opinions here. Chroma calls this `EmbeddingFunction` should we align to that? Are there other references that we should look at?

Reply:
EmbeddingFunction makes more sense if you're "calling" it on some text/chunks.  The user treats this as a model, so I think it should be named as such.  Although, now that we have embed_chunks as the only function of it, we could change it to be EmbeddingFunction as chromaDB does and make it callable.  The issue with that is that something that simple shouldn't require a chunk input, it should just be text.  I've made the change in a new branch [here](https://github.com/peachkeel/ragna/tree/callable-embedding-models) so you can have a look